### PR TITLE
Gtid: feature gate __cmp__ since it's python2.

### PR DESCRIPTION
**Pull request by darnuria**
I can also remove it if python 2.7 support is no longer needed.

### CODE REVIEW

File: pymysqlreplication/gtid.py [hunk 1]
```diff
@@ -246,6 +247,10 @@ def decode(cls, payload):
             for x in intervals])))
 
     def __cmp__(self, other):
+        """Compare two Gtid only work in python 2.7 since it rely on cmp."""
+        if version_info[0] >= 3:
+            raise NotImplemented
+
```
Comment (darnuria) [inline pymysqlreplication/gtid.py:253]: Found while doing typing with mypy:

```
pymysqlreplication/gtid.py:254: error: Name "cmp" is not defined  [name-defined]
pymysqlreplication/gtid.py:255: error: Name "cmp" is not defined  [name-defined]
```

**Comment by julien-duponchelle** [comment]
Oh nice!

**Comment by darnuria** [comment]
@julien-duponchelle so it raise one big question: what can I do? Remove if, if python2 is dropped I think it's for the best (not an expert in python). Or I can feature gate it (may need a test to avoid footgunning) but I don't have a python2 anymore. XD

**Comment by julien-duponchelle** [comment]
I'm happy to drop python 2 !

**Comment by darnuria** [comment]
> I'm happy to drop python 2 !

Done, sorry for the lag busy days! ;)

**Comment by julien-duponchelle** [comment]
Sorry for the late response but I merged it and I'm in the process of doing a release

**Pull request merged**